client/network/src/request_responses.rs

```diff
@@ -344,41 +369,17 @@ impl NetworkBehaviour for RequestResponsesBehaviour {
 						)
 					}
 				},
-			FromSwarm::DialFailure(DialFailure { peer_id, error, handler }) =>
-				for (p_name, p_handler) in handler.into_iter() {
-					if let Some((proto, _)) = self.protocols.get_mut(p_name.as_str()) {
-						proto.on_swarm_event(FromSwarm::DialFailure(DialFailure {
-							peer_id,
-							handler: p_handler,
-							error,
-						}));
-					} else {
-						log::error!(
-						  target: "sub-libp2p",
-						  "on_swarm_event/dial_failure: no request-response instance registered for protocol {:?}",
-						  p_name,
-						)
-					}
+			FromSwarm::DialFailure(e) =>
+				for (p, _) in self.protocols.values_mut() {
```

Comment:
> this assumes that for every connection we register all protocols.

Sounds correct

> not sure why handler.into_iter() was used before since the logic seems equal in both cases to me

I don't know either but it looks right what has changed here.